### PR TITLE
AndroidMediaNdkCapture pixel format enhancement

### PR DESCRIPTION
**Pull request by warped-rudi**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/videoio/src/cap_android_mediandk.cpp [hunk 1]
```diff
@@ -51,9 +61,9 @@ class AndroidMediaNdkCapture : public IVideoCapture
 public:
     AndroidMediaNdkCapture():
         sawInputEOS(false), sawOutputEOS(false),
-        frameStride(0), frameWidth(0), frameHeight(0), colorFormat(0),
-        videoWidth(0), videoHeight(0),
-        videoFrameCount(0),
+        frameStride(0), frameWidth(0), frameHeight(0),
+        colorFormat(COLOR_FormatUnknown), fourCC(FOURCC_BGR),
```
Comment (asmorkalov) [inline modules/videoio/src/cap_android_mediandk.cpp:65]: Attention: default output become BGR for Android too.

**Review by asmorkalov — APPROVED** [review]

**Comment by asmorkalov** [comment]
@warped-rudi Thanks a lot for the contribution. The BGR->RGB is breaking change. I'll will discuss it with core team and return back. We also have NativeView class based on VideoCapture. Please add color space setter there to prevent wrong colors.

**Comment by warped-rudi** [comment]
I think the cause of the problem is the following: [From this code here](https://github.com/opencv/opencv/blob/d9a139f9e85a17b4c47dbca559ed90aef517c279/modules/videoio/src/cap_android_mediandk.cpp#L165) I gather, that the intention is to produce BGR. The reason why we get something like RGB instead is, that the source format is not NV21, but NV12. This is backed by [line 514](https://github.com/opencv/opencv/blob/d9a139f9e85a17b4c47dbca559ed90aef517c279/modules/videoio/src/cap_android_mediandk.cpp#L514). AndroidMediaNdkVideoWriter basically uses the same hardware accelerator and it's quite unlikely, that different formats get used for encoding and decoding.

**Comment by warped-rudi** [comment]
> We also have NativeView class based on VideoCapture. Please add color space setter there to prevent wrong colors.

I can't find this class. Where is it?

**Comment by asmorkalov** [comment]
https://github.com/opencv/opencv/blob/4.x/modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java

**Comment by warped-rudi** [comment]
O.K. I see. But this code uses a camera, which means that it instantiates 'AndroidCameraCapture' instead of 'AndroidMediaNdkCapture'. Thus it's unaffected by this change set.

However, with #26627 in place, [line 184](https://github.com/opencv/opencv/blob/aa52dafc90723c996fd6cff6b6c1a9b12d25f58d/modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java#L184) can be changed to directly capture BGR4 avoiding the conversion step in  [line 187](https://github.com/opencv/opencv/blob/aa52dafc90723c996fd6cff6b6c1a9b12d25f58d/modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java#L187).

If you modify the code to read from a file instead from a camera, then you will see the bug reported in #26616 and the effect of this patch.

**Comment by asmorkalov** [comment]
Please also take a look on https://github.com/opencv/opencv/blob/4.x/samples/android/video-recorder/src/org/opencv/samples/recorder/RecorderActivity.java. Looks like we need to set proper output fourcc and drop cvtColor before imwrite.

**Comment by komakai** [comment]
My understanding is that that video capturers should output frames in BGR format, so it appears there is a bug in my original implementation for Android.
It should be noted that the macOS video capturer outputs frames in RGB format so this may have caused me some confusion.

**Comment by warped-rudi** [comment]
@asmorkalov : The recorder sample is affected. I will have look at it on weekend...

**Pull request merged**